oracle-linux-image-tools/bin/mnt-img.sh

```diff
@@ -190,18 +194,50 @@ fi
 mapped_devices=$(kpartx -l "$LPDEVICE" | grep "$LPDEVICE" | awk '{print $1}')
 ct=1
 for d in $mapped_devices; do
-  mkdir -p "$MOUNT_POINT"/$ct
   file -sL /dev/mapper/"$d"
   # mount ext3, ext4, btrfs, xfs partitions
   if file -sL /dev/mapper/"$d" | egrep -q -i "ext3|ext4|btrfs|xfs"; then
+     mkdir -p "$MOUNT_POINT"/$ct
      mount /dev/mapper/"$d" "$MOUNT_POINT"/$ct
      echo "MOUNTED DIR:/dev/mapper/$d $MOUNT_POINT/$ct" |tee -a $WORK_HOME/"$ID"
      ct=$((ct+1))
+  elif file -sL /dev/mapper/"$d" | egrep -q -i "lvm2"; then
+     echo "LVM detected"
+     guest_vg=1
+     pvscan --cache /dev/mapper/"$d"
   else
      echo "/dev/mapper/$d not mounted - Unknown filesystem."
   fi
-
 done
+
+if [[ ${guest_vg} == 1 ]]; then
+  # Image has LVM
+  # This only works for simple setups (where the whole VG is on the device)
+  # Scan for Volume Groups
+  all_vgs=$(vgscan | grep "Found volume group" | sed -e 's/^[^"]*"\([^"]*\)".*$/\1/')
+  for vg in ${all_vgs}; do
+    # Only consider new VGs
+    if ! grep -w -q "${vg}" <<<${host_vgs}; then
+      # Ensure the VG is active and register it
+      vgchange -ay "${vg}"
+      echo "LVM GROUP:${vg}" | tee -a "${WORK_HOME}/${ID}"
+      lvscan >/dev/null
+      # Search for filesystems in the VG
+      for fs in "/dev/${vg}"/*; do
+        file -sL "${fs}"
+        if file -sL "${fs}" | egrep -q -i "ext3|ext4|btrfs|xfs"; then
```

Comment:
Why are you duplicating `file -sL` on lines 227 and 228?

Reply:
First one to display the file type and second one to process the output.
We could save and reuse the output, but that doesn't make the code much more readable and the `file` operation doesn't use much resources...

Reply:
Is there a need to display the file type? What value does that output have at this point?

Reply:
Mainly informative / debugging. We will miss cleanup is we don't identify a FS.
I moved that to the `else` part of the if to only display unidentified FS.
88fab62